backend/danswer/db/models.py

```diff
@@ -556,13 +556,14 @@ class EmbeddingModel(Base):
     index_name: Mapped[str] = mapped_column(String)
 
     # New field for cloud provider relationship
-    cloud_provider_id: Mapped[int | None] = mapped_column(
-        ForeignKey("embedding_provider.id")
+    provider_type: Mapped[EmbeddingProvider | None] = mapped_column(
+        ForeignKey("embedding_provider.provider_type")
```

Comment:
should this be nullable?

Reply:
In my opinion, yes! Null for self-hosted models. Can make more explicit by adding a `nullable`